### PR TITLE
feat: support for DN in venafi_certificate

### LINKED ISSUE #108: Adding O, OU, S, L to the certificate object
**BUSINESS PROBLEM**

Requesting for O, OU, S, L to the certificate object to be added to terraform, as the "recommended settings" are not configured for all "Issuing templates" as we have set value for O, OU, S, L to the certificate object which is requiring us to double up and reapply these values to "recommended settings"

**PROPOSED SOLUTION**

Requesting for O, OU, S, L to the certificate object to be added for the terraform option rather than only replying on "recommended settings"

**CURRENT ALTERNATIVES**

Manually editing each template to add "recommended settings" 

**VENAFI EXPERIENCE**

1 year +

### PR DESCRIPTION
**Pull request by marcos-albornoz**
Closes #108

### CODE REVIEW

File: venafi/resource_venafi_certificate_test.go [hunk 1]
```diff
@@ -465,6 +533,204 @@ func TestVaasSignedCert(t *testing.T) {
...
+// |                          | Terraform venafi_certificate      | CIT configuration                                  | Recommended Settings | Resulting Certificate             |
+// +==========================+===================================+====================================================+======================+===================================+
+// | Organization(O)          |                                   | Venafi Inc.                                        | Venafi Inc.          |                                   |
+// | Organizational Units(OU) | Professional Services;Sales       | Customer Support;Professional Services;Engineering | Customer Support     | *FAIL given Sales is not an       |
+// | Locality(L)              | Merida                            | Salt Lake,Merida                                   | Salt Lake            | accepted value by the CIT         |
+// | State(ST)                | Yucatan                           | Utah; Yucatan                                      | Utah                 |                                   |
```
Comment (luispresuelVenafi) [inline venafi/resource_venafi_certificate_test.go:601]: Ah, the little touch of Yucatán, MX, in here. :1st_place_medal:

**Review by EduardoVV — APPROVED** [review]

**Review by luispresuelVenafi — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## 0.21.1 (September 10th, 2024)
```
Comment (rvelaVenafi) [inline CHANGELOG.md:1]: version 0.21.1 does not exist. If these changes are already in master, all of them should be documented under next release which is 0.22.0

Comment (rvelaVenafi) [reply CHANGELOG.md:1]: Also your changes are missing here (adding DN attributes and all of that)

File: venafi/resource_venafi_certificate.go [hunk 3]
```diff
@@ -531,6 +562,33 @@ func enrollVenafiCertificate(ctx context.Context, d *schema.ResourceData, cl end
...
+	}
+	locality := d.Get("locality").(string)
+	if locality != "" {
+		req.Subject.Locality = []string{locality}
+	}
+	org := d.Get("organization").(string)
+	if org != "" {
+		req.Subject.Organization = []string{org}
+	}
+	orUnitsNum := d.Get("organizational_units.#").(int)
```
Comment (rvelaVenafi) [inline venafi/resource_venafi_certificate.go]: typo here `orUnitsNum` should be `orgUnitsNum`

File: venafi/resource_venafi_certificate.go [hunk 4]
```diff
@@ -531,6 +562,33 @@ func enrollVenafiCertificate(ctx context.Context, d *schema.ResourceData, cl end
...
+	if orUnitsNum > 0 {
```
Comment (rvelaVenafi) [inline venafi/resource_venafi_certificate.go]: No need to do this check, if orUnitsNum is 0, then the for loop is ignored

File: venafi/resource_venafi_certificate.go [hunk 5]
```diff
@@ -531,6 +562,33 @@ func enrollVenafiCertificate(ctx context.Context, d *schema.ResourceData, cl end
...
+		for i := 0; i < orUnitsNum; i++ {
+			key := fmt.Sprintf("organizational_units.%d", i)
```
Comment (rvelaVenafi) [inline venafi/resource_venafi_certificate.go]: You don't need to iterate the slice based on the index.
You can assert the type of the slice and then assert the element's type when iterating it:

```
orgUnits := d.Get("organizational_units").([]interface{})
```
Now you have a list of interfaces ready to iterate:
```
for i, item := range orgUnits{
    itemStr := item.(string)
    req.Subject.OrganizationalUnit = append(req.Subject.OrganizationalUnit, itemStr)
}
```

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: venafi/resource_venafi_certificate.go (hunk 4 above)
Comment (marcos-albornoz) [reply venafi/resource_venafi_certificate.go]: done

File: venafi/resource_venafi_certificate.go (hunk 5 above)
Comment (marcos-albornoz) [reply venafi/resource_venafi_certificate.go]: done

File: venafi/resource_venafi_certificate.go (hunk 3 above)
Comment (marcos-albornoz) [reply venafi/resource_venafi_certificate.go]: done

**Review by luispresuelVenafi — APPROVED** [review]

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**